workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowFolderWatcher.java

```diff
@@ -139,8 +139,13 @@ private void addWorkflow(Path workflowFile) throws IOException, ProcessingExcept
       return;
     }
     Workflow workflow = SwadlParser.fromYaml(workflowFile.toFile());
-    workflowEngine.deploy(workflow);
-    deployedWorkflows.put(workflowFile, workflow.getId());
+    Object instance = workflowEngine.parseAndValidate(workflow);
+    if (workflow.isToPublish()) {
+      workflowEngine.deploy(workflow, instance);
+    } else if (deployedWorkflows.get(workflowFile) != null && deployedWorkflows.get(workflowFile).getRight()) {
```

Comment:
Does this mean, if we have a workflow already deployed, and we update the workflow file and we put it as draft, the original one will get undeployed? Is it the behaviour we really want? 
For me, the draft mode should not change the already deployed workflows. The management api #127 should be used for that.